Eastern-Research-Group/csb-rebate-forms-app
# Feature/integrate close out form

## Related Issues:
* CSBAPP-5

## Main Changes:
* Ongoing work to support the Close-Out form – mostly setting up the server app's API endpoints for fetching and updating Close-Out form data from Formio, and stubbing out the BAP query function that will fetch BAP data from an accepted Payment Request form submission to then inject into a brand new Close-Out form submission.
* Also fixes a bug that slipped into the payment request form page that prevented the form from showing the latest submitted form fields when the "Save" or "Next" buttons were clicked (data was submitted to Formio, but the form didn't properly render it, due to a typo in the react-query mutation's onSuccess callback updating data in the wrong cache) – this was just a copy/paste bug introduced in a recent pull request. 

## Steps To Test:
1. Run app – all should still function as normal.
2. A new route exists at `/close-out/:id` (__NOTE:__ still just a placeholder page for now)

## TODO:
- [ ] Flesh out the the Close-Out form route component to display the Close-Out form component
